### PR TITLE
Feature/laravel jobs permission checking

### PR DESCRIPTION
**Pull request by supertassu**
This depends on #33 for MW api base and is part of master task #59.

### CODE REVIEW

**Review by supertassu — COMMENTED** [review]
Noting here that this does not do anything to banned users (yet)

**Review by supertassu — COMMENTED** [review]
I think that this is ready for review. I have a feeling that I've forgotten something but I can't seem to remember what and it probably wasn't anything major. @dqwiki Please double-check that the logic itself is correct and please ask if you don't understand something.

File: app/Jobs/WikiPermission/BaseWikiPermissionJob.php [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+        return $groups;
+    }
+
+    /**
+     * @param bool $exists true if this user exists on the wiki this job is querying
+     */
+    public function updateDoesExist(bool $exists)
+    {
+        $wikis = explode(',', $this->user->wikis ?? '');
```
Comment (dqwiki) [inline app/Jobs/WikiPermission/BaseWikiPermissionJob.php:83]: This can fail. $wikis can have a Null value, a single value, or multiple values.

**Comment by supertassu** [comment]
... and we need to deal with changes in #136.

I'd much rather have the `user` permission assigned to everybody who exists on that wiki and only give access to users with `admin` permission to make dealing with #140 much easier

**Review by dqwiki — COMMENTED** [review]
Sorry, forgot to push this

**Comment by dqwiki** [comment]
> ... and we need to deal with changes in #136.
> 
> I'd much rather have the `user` permission assigned to everybody who exists on that wiki and only give access to users with `admin` permission to make dealing with #140 much easier

Absolutely fine with that, my hacks on 136 can be changed up as needed.

File: app/Jobs/WikiPermission/BaseWikiPermissionJob.php (hunk 1 above)
Comment (dqwiki) [reply app/Jobs/WikiPermission/BaseWikiPermissionJob.php:83]: I lies.

File: app/User.php [hunk 2]
```diff
@@ -2,7 +2,10 @@
 
 namespace App;
 
+use App\MediawikiIntegration\WikiApiUrls;
```
Comment (dqwiki) [inline app/User.php]: This doesn't seem to exist and also errors on my terminal as not found.

**Review by dqwiki — CHANGES_REQUESTED** [review]
Switch to request changes

**Review by dqwiki — APPROVED** [review]

**Pull request merged**